apps/src/blockly/addons/functionEditor.js

```diff
@@ -23,80 +111,237 @@ export default class FunctionEditor {
   // TODO
   refreshParamsEverywhere() {}
 
-  // TODO
-  openEditorForFunction(procedureBlock, functionName) {}
+  /**
+   * Show the given procedure in the function editor. Either load from
+   * the procedure workspace if it already exists, or create a new block.
+   * @param {Procedure} procedure The procedure to show.
+   */
+  showForFunction(procedure) {
```

Comment:
This is a continuation of the last comment -- what are your thoughts on whether we care about matching the names of the functions we use to the names of functions used in the codebase at this point? `openEditorForFunction` and `openWithNewFunction` were placeholders for previous implementations.

<img width="594" alt="Screenshot 2023-08-21 at 9 39 40 AM" src="https://github.com/code-dot-org/code-dot-org/assets/26844240/91ba8ea3-5486-4f3a-8312-c90742f7406c">

Reply:
Same as before, reason to keep them consistent is because we would need to change the cdo version in order to rename them right now.

Reply:
Edit: realized we renamed this one and that's ok because we only called it within google blockly code

Reply:
In general, if the rename would only impact new code and we have a better name now, then I think we can go for it. If it would require changing a cdo name, I think we keep it the same.

Reply:
Sorry, not very clear explanation on my part. If we are accessing the function via `Blockly.<function>` or `Blockly.<classname>.<function>` we will have to be very careful in order to rename it. The reason is when we call a function prefixed with `Blockly`, in general we expect it to work for either CDO or Google Blockly (we are actually calling via `cdoBlocklyWrapper` or `googleBlocklyWrapper`. 

So for `hideIfOpen`, I don't think we should rename it. [This function call](https://github.com/code-dot-org/code-dot-org/blob/81516efb2105641ee5f09f50874b187f98d302ba/apps/src/StudioApp.js#L839) needs to work for both google and cdo blockly. If we wanted to rename it we would need to update the function editor in the cdo blockly repo, which feels unnecessary.

For `openWithNewFunction` and `openEditorForFunction`, are we planning to use `functionEditor` for both functions and behaviors? If so, we probably need to change the name back because of the call you linked above. Otherwise, the only place the current implentation is calling `showForFunction` is [here](https://github.com/code-dot-org/code-dot-org/pull/53114/files#diff-8c6c333b74263db24d773ecbfd5cd4d66708e6cdd5ec730094f783cf9f501937R99), which is guaranteed to always use the google blockly wrapper because it's a google blockly specific file.

Reply:
for now I have set `hideIfOpen` to just call `hide`

Reply:
Thanks, this makes sense! In the previous implementation, the behavior editor and function editor were both instances of the function editor class:
```
const behaviorEditor = (Blockly.behaviorEditor = new Blockly.FunctionEditor
```
My high-level thought is that our options are to 1) use the function editor for both functions and behaviors or 2) have a BehaviorEditor class that extends the FunctionEditor class, and I imagine we'd start with #1 and then pivot to #2 if it feels necessary?

Reply:
yeah that makes sense to me. When we get there we may need to rename the functions then.